app/src/main/java/org/mozilla/focus/utils/UrlUtils.java

```diff
@@ -123,7 +123,10 @@ public static boolean isInternalErrorURL(final String url) {
         return "data:text/html;charset=utf-8;base64,".equals(url);
     }
 
-    public static boolean urlsMatchExceptForTrailingSlash(final @NonNull String url1, final @NonNull String url2) {
+    public static boolean urlsMatchExceptForTrailingSlash(final String url1, final String url2) {
+        if (url1 == null || url2 == null) {
```

Comment:
It may be unexpected that this returns false if they're both null – can you add a javadoc comment?